### PR TITLE
Add GitHub Actions CI for Windows and Linux

### PR DESCRIPTION
**Pull request by Margen67**
Supersedes #71 due to branch name.

#
No macOS builds since they still fail: https://github.com/Margen67/OpenAG/runs/550373800#step:4:113

### CODE REVIEW

**Comment by Margen67** [comment]
@YaLTeR Also, VS2019 fails: https://github.com/Margen67/OpenAG/runs/550715566

**Comment by YaLTeR** [comment]
> @YaLTeR Also, VS2019 fails: https://github.com/Margen67/OpenAG/runs/550715566

You need to pass `-A Win32` to CMake when generating the project. You can pass it on VS 2017 too, it shouldn't break it.

**Comment by YaLTeR** [comment]
> No macOS builds since they still fail: https://github.com/Margen67/OpenAG/runs/550373800#step:4:113

What is this for: `sudo xcode-select -s /Applications/Xcode_11.app/Contents/Developer`?

**Comment by Margen67** [comment]
> 
> 
> > @YaLTeR Also, VS2019 fails: https://github.com/Margen67/OpenAG/runs/550715566
> 
> You need to pass `-A Win32` to CMake when generating the project. You can pass it on VS 2017 too, it shouldn't break it.

That fixed `windows-latest`.
> 
> 
> > No macOS builds since they still fail: https://github.com/Margen67/OpenAG/runs/550373800#step:4:113
> 
> What is this for: `sudo xcode-select -s /Applications/Xcode_11.app/Contents/Developer`?

It was to try a slightly older version of Xcode, but it didn't work.

**Comment by YaLTeR** [comment]
Can you send me the output without the `xcode-select` line?

**Comment by Margen67** [comment]
> 
> 
> Can you send me the output without the `xcode-select` line?

@YaLTeR https://github.com/Margen67/OpenAG/runs/551006549?check_suite_focus=true#step:4:113

**Comment by YaLTeR** [comment]
So, the macOS failure seems to be caused by the removal of 32-bit support in macOS Catalina. Looking at [the list of installed SDKs](https://github.com/actions/virtual-environments/blob/master/images/macos/macos-10.15-Readme.md#installed-sdks), they only have the 10.15 SDK installed. So I suppose we can just keep macOS on Travis where they have an older version.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        name: openag-${{ runner.os }}-${{ matrix.configuration }}
+        path: build\${{ matrix.configuration }}\client.dll
+
+  build-linux:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [Release, Debug]
+        gxx-ver: [6, 7, 8]
```
Comment (YaLTeR) [inline .github/workflows/CI.yml:52]: Nitpick: this is better called gcc-ver. Also, do they have gcc 9 and 10 in that repo?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        submodules: recursive
+    - name: Install packages
+      run: |
+        sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+        sudo apt-get update
+        sudo apt-get -y install g++-${{ matrix.gxx-ver }} g++-${{ matrix.gxx-ver }}-multilib linux-libc-dev-i386-cross mesa-common-dev ninja-build rapidjson-dev
+        mkdir build
+        echo "::set-env name=CC::gcc-${{ matrix.gxx-ver }}"
+        echo "::set-env name=CXX::g++-${{ matrix.gxx-ver }}"
+    - name: Cmake generate
```
Comment (YaLTeR) [inline .github/workflows/CI.yml:65]: Nitpick: it's CMake.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      working-directory: build
+      run: cmake -G Ninja ..
```
Comment (YaLTeR) [inline .github/workflows/CI.yml:67]: Missing `-DCMAKE_BUILD_TYPE`

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      with:
+        submodules: recursive
+    - name: Install packages
+      run: |
+        sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+        sudo apt-get update
+        sudo apt-get -y install g++-${{ matrix.gxx-ver }} g++-${{ matrix.gxx-ver }}-multilib linux-libc-dev-i386-cross mesa-common-dev ninja-build rapidjson-dev
+        mkdir build
+        echo "::set-env name=CC::gcc-${{ matrix.gxx-ver }}"
+        echo "::set-env name=CXX::g++-${{ matrix.gxx-ver }}"
```
Comment (YaLTeR) [inline .github/workflows/CI.yml:64]: Probably cleaner to set these with `env:` on the CMake generation step.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (YaLTeR) [inline .github/workflows/CI.yml:52]: How hard would it be to also add clang? It's tested in Travis currently.

Comment (Margen67) [reply .github/workflows/CI.yml]: Ubuntu Bionic doesn't have 9 or 10.

**Comment by Margen67** [comment]
@YaLTeR clang fails: https://github.com/Margen67/OpenAG/runs/559991669#step:5:7

**Comment by YaLTeR** [comment]
There's some error in the .yml because CMake thinks it's GCC and not Clang:
```
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
```

**Comment by Margen67** [comment]
@YaLTeR This is ready to merge.

**Comment by YaLTeR** [comment]
Thanks a lot for working on this!

**Pull request merged**